validphys2/src/validphys/pdfplots.py

```diff
@@ -601,6 +602,25 @@ def legend(self, flstate):
         )
 
 
+class BandPDFPlotterBC(BandPDFPlotter):
+    def __init__(self, *args, unpolarized_bcs, boundary_xplotting_grids, **kwargs):
+        self.unpolarized_bcs = unpolarized_bcs
+        self.boundary_xplotting_grids = boundary_xplotting_grids
+        super().__init__(*args, **kwargs)
+
+    def draw(self, pdf, grid, flstate):
+        xgrid = grid.xgrid
+        ax = flstate.ax
+        # TODO: Maybe not always select the first element in the list
```

Comment:
I'm a bit confused about this TODO. If the other members are not used, why are they there at all?

If `boundary_xplotting_grids` should be the same size as the list of PDF being plotted (it's not clear to me) you can make it an iterator in the constructor and get the next every time you go through draw (you should pass through here as many times as pdfs are there to be drawn if I'm not mistaken)

Reply:
This has been just taken into in [88a7b25](https://github.com/NNPDF/nnpdf/pull/1979/commits/88a7b25f3adede4b00111c0d008354693a73f8ad) but I could briefly explain how this ended up there. Usually, when comparing two fits, these two fits are produced with the same BC meaning that `unpolarized_bcs` will contain two elements with the exact same thing. But all the variations should now be taken into account in [88a7b25](https://github.com/NNPDF/nnpdf/pull/1979/commits/88a7b25f3adede4b00111c0d008354693a73f8ad).